### PR TITLE
Books suggestion creation route

### PR DESCRIPTION
**Pull request by mnmallea**
## Summary
- Added `POST /book_suggestions` route.

## Trello card
[Link](https://trello.com/c/Bg6DXOS7/26-crear-ruta-para-creaci%C3%B3n-de-un-booksuggestion)

### CODE REVIEW

File: app/controllers/book_suggestions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    book_suggestion = BookSuggestion.new(book_suggestion_params)
+
+    if book_suggestion.save
+      render json: book_suggestion, status: :created
+    else
+      render json: book_suggestion.errors, status: :unprocessable_entity
+    end
+  end
+
+  def book_suggestion_params
```
Comment (lucasVoboril) [inline app/controllers/book_suggestions_controller.rb:11]: This is a private method :wink:

File: spec/factories/book_suggestions.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 FactoryBot.define do
   factory :book_suggestion do
     books_attributes
-    synopsis { Faker::String.random }
+    synopsis { Faker::String.random.gsub("\u0000", '') }
```
Comment (lucasVoboril) [inline spec/factories/book_suggestions.rb]: Why you did this? 
```suggestion
    synopsis { Faker::String.random.gsub("\u0000", '') }
```

File: spec/controllers/book_suggestions_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+describe BookSuggestionsController do
+  describe 'POST #create' do
+    subject(:http_response) { post :create, params: { book_suggestion: book_suggestion } }
+
+    context 'when posting a new book suggestion' do
+      let(:book_suggestion) do
```
Comment (lucasVoboril) [inline spec/controllers/book_suggestions_controller_spec.rb]: :thinking: you can use FactoryBot.attributes_for(:book_suggestion) to do this :wink: [chek it](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#using-factories)

File: spec/controllers/book_suggestions_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+      it { is_expected.to have_http_status :created }
+
+      it 'responses with a json' do
+        expect(response.content_type).to eq 'application/json'
+      end
+    end
+
+    context 'when posting an invalid book suggestion' do
+      let(:book_suggestion) do
```
Comment (lucasVoboril) [inline spec/controllers/book_suggestions_controller_spec.rb]: Same here

**Review by lucasVoboril — CHANGES_REQUESTED** [review]

File: app/controllers/book_suggestions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+class BookSuggestionsController < ApplicationController
+  def create
+    book_suggestion = BookSuggestion.new(book_suggestion_params)
+
+    if book_suggestion.save
```
Comment (leoncalermo) [inline app/controllers/book_suggestions_controller.rb]: we try to not use the if and else thing.. maybe you could  use
```
return render json: book_suggestion, status: :created if book_suggestion.save
render json: book_suggestion.errors, status: :unprocessable_entity
```

File: spec/factories/books_attributes.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  trait :books_attributes do
```
Comment (leoncalermo) [inline spec/factories/books_attributes.rb]: i dont understand why a whole factory is only a trait 🤔

**Review by leoncalermo — CHANGES_REQUESTED** [review]

File: spec/factories/books_attributes.rb (hunk 6 above)
Comment (mnmallea) [reply spec/factories/books_attributes.rb]: I've used this trait to share common attributes between books and book suggestions

**Review by lucasVoboril — APPROVED** [review]

**Pull request merged**